src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -85,11 +85,16 @@ def update_building
     render json: workers
   end
 
-  def gethistory(key, range, cache = 1)
-    cachekey = key + "-#{range}"
-    Rails.cache.delete(cachekey, shared: true) unless cache
-    Rails.cache.fetch(cachekey, expires_in: (range.to_i * 3600) / 150, shared: true) do
-      hash = StatusHistory.history_by_key_and_hours(key, range)
+  def gethistory(key, range, end_date, cache = 1)
+    if end_date.empty?
```

Comment:
I wanted to check wether `end_date` was `''` or not, for deciding when to use the cache.